uvicorn/config.py

```diff
@@ -132,6 +133,57 @@ def create_ssl_context(
     return ctx
 
 
+def is_dir(path: Path) -> bool:
+    try:
+        if not path.is_absolute():
+            path = path.resolve()
+        return path.is_dir()
+    except OSError:
+        return False
+
+
+def resolve_reload_patterns(
+    patterns_list: List[str], directories_list: List[str]
+) -> Tuple[List[str], List[Path]]:
+
+    directories: List[Path] = list(set(map(Path, directories_list.copy())))
```

Comment:
writing this without thinking too much or having read the rest, would a deepcopy be better or not ?

Reply:
No we don't need a deep copy here as we can keep the path elements. I just wanted to decouple the list inside from the outside as I still need the outside one to check what was passed to the config.